File: problems/power-of-two/recursion.cs
public class Solution {
    public bool IsPowerOfTwo(int n) {
        if (n == 0) return false;
        if (n == 1) return true;
        if (n == 3) return false;
        return n % 2 == 0 && IsPowerOfTwo(n / 2);
    }
}

File: problems/count-and-say/recursion.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"

        digits = self.countAndSay(n - 1)
        rle = self.getRunLengthEncoding(digits)
        return rle
        
    def getRunLengthEncoding(self, digits: str) -> str:
        frequencies = [ [ digits[0], 1 ] ]

        for i in range(1, len(digits)):
            digit = digits[i]
            # If the last frequency pair in the array is equal to the current digit
            # increment the value
            if frequencies[-1][0] == digit:
                frequencies[-1][1] += 1
            else:
                # Add a new frequency for a new digit
                frequencies.append([digit, 1])
        
        output = ""
        for freq in frequencies:
            output += str(freq[1]) + freq[0]
        #print(output)

        return output
